### DISCUSSION TITLE
Scalable / resizeable UI

### DISCUSSION THREAD

**Discussion by raenji-sk** [terjeio/ioSender#97] [Ideas]
I wonder how difficult it would be to add a scaling and resizing option for the UI. The UI is pretty small on a 21" fullHD LCD.
To be precise I use it with a touch screen LCD, so the buttons are difficult to find and press quickly.
Also the Jog/Macros... panels on the left could be a bit biggger. I used bCNC for a long time and it had a double sized icons options which was nice and helpful when used with a touchscreen.
I've seen you have some UI improvements in the roadmap, but there is no mention about scaling.

**Comment by terjeio** [comment]
Resizing is already implemented, and size can be saved across sessions if enabled in _Settings: app_. Some UI-elements (containers) are scaled and some of them hidden when the size is too small. E.g. the _Program limits_ container will not be shown when more than 4 (?) axes are displayed in the DRO unless the window is resized to be large enough for it.

> To be precise I use it with a touch screen LCD

So no keyboard? What is the size and resolution?

> Also the Jog/Macros... panels on the left could be a bit biggger.

Is your controller running grblHAL? If so and you have acces to a keyboard keyboard jogging is an option for you?

The macro buttons in the macro panel are duplicated in the toolbar and are a bit larger there, still too small?

> I used bCNC for a long time and it had a double sized icons options which was nice and helpful when used with a touchscreen.

I'll have to check how that works.

**Reply by raenji-sk** [reply]
> size can be saved across sessions if enabled in Settings: app.

But this only works for the whole app right? not individual containers?

> Is your controller running grblHAL? If so and you have acces to a keyboard keyboard jogging is an option for you?

I am running Phils Teensy breakout board with latest grblHAL, compiled about 2 weeks ago.

> So no keyboard? What is the size and resolution?

Using a wireless keyboard most of the time, LCD is 21" and 1920x1080 resolution

>The macro buttons in the macro panel are duplicated in the toolbar and are a bit larger there, still too small?

They could be a bit bigger, yes.

Generally I prefer the keyboard approach, but sometimes when I am close to the screen, or forget the keyboard on the other side of the shop, it would be faster to press a button there.
Jogging is no problem with a keyboard, but perhaps buttons like the macros on top of the UI, zeroing of the axes in DRO, cycle start, feed hold/resume would benefit from a larger size, maybe even some coloring, like green for start, yellow for pause, red for stop.
Just my personal opinion, but I think the middle of the UI is unnecessarily big (program/3d/console view) and could be smaller without limiting its function and on the other hand the controls could be larger and make them easier to press be it with a touch screen or a mouse. And yes I know, keyboard shortcuts or big physical buttons (like for emergency stop) are best.

I do not know whether you're familiar with UCCNC software, but I personally like their UI. Wanted to give their HW/SW a try, but then I discovered grblHAL and decided to give it a go as it looks, and so far also works, really good.
Anyway here is a link to my Dropbox with some comparison screenshots of the UI of ioSender and UCCNC on a 1920x1200 monitor, so almost the same as I use for the CNC. [https://www.dropbox.com/sh/b8yt4az4xp5ofy5/AACRwAULeDxFnjs0XpwheZ8ta?dl=0](url)

To sum it up, it is not something that I can not live without, so if it would take a lot of time and effort to implement or you personally do not see any benefit in it, then of course I will have no prbolem with that (and will keep using ioSender ;) )

**Reply by terjeio** [reply]
> I do not know whether you're familiar with UCCNC software

I am not, IMO the UI looks a bit cluttered to me.

As for the UI one of my ideas way back was to have the main window as a canvas where UI components could be dropped in and arranged by the user. A bit like what [foobar 2000](https://www.foobar2000.org/) allows?

> To sum it up, it is not something that I can not live without, so if it would take a lot of time and effort to implement or you personally do not see any benefit in it...

I did not publish ioSender for my personal benefit, had I known how much work it has turned out to be I might have kept it to myself ;-)

Feedback is appreciated and taken note of, and suggested changes/features may eventually end up in the app.  Be patient...

**Reply by raenji-sk** [reply]
> IMO the UI looks a bit cluttered to me.

Yes, it is a bit overengineered, but it works just fine after you get used to it. And the buttons are about right size so can be used with a touchscreen comfortably. Not saying ioSender should look like that, just something to compare it to.
Also they do have the option of creatig custom screen sets, by defining buttons and their function via XMLs.

>Be patient...

I will, do not worry. Migth have some more suggestions after using it for a longer time. But in general it is my main gcode sender as of now.

**Comment by phil-barrett** [comment]
> I did not publish ioSender for my personal benefit, had I known how much work it has turned out to be I might have kept it to myself ;-)

Well, I am glad you did.  It is the best sender out there by far.

**Comment by terjeio** [comment]
I have increased the minimum window size a bit. Some buttons are also made a bit bigger, the most in the jog flyout. Available from the [edge version download](http://www.io-engineering.com/downloads) for now.
